cmake/cyclonedds/CMakeLists.txt

```diff
@@ -70,7 +70,13 @@ function(fetch_and_install name)
     endforeach()
     string( REPLACE ";" " " ADDITIONAL_CMAKE_FLAGS "${ADDITIONAL_CMAKE_FLAGS}")
 
-    execute_process(COMMAND ${CMAKE_COMMAND} -G "${CMAKE_GENERATOR}" "-DCMAKE_INSTALL_PREFIX=${INSTALL_DIR}" "${ADDITIONAL_CMAKE_FLAGS}" "${TOOLCHAIN_FILE}" "${EXTRA_CMAKE_ARGS}"  "${SOURCE_DIR}"
+    set(CYCLONE_BUILD_ARGS "${CMAKE_GENERATOR}" "-DCMAKE_INSTALL_PREFIX=${INSTALL_DIR}" "${EXTRA_CMAKE_ARGS}" "${ADDITIONAL_CMAKE_FLAGS}" "${SOURCE_DIR}")
+
+    if(ADDITIONAL_CMAKE_FLAGS STREQUAL "")
+        list(APPEND CYCLONE_BUILD_ARGS ${ADDITIONAL_CMAKE_FLAGS})
+    endif()
+
+    execute_process(COMMAND ${CMAKE_COMMAND} -G ${CYCLONE_BUILD_ARGS}
```

Comment:
This is a bit brittle. `-G` and `"${CMAKE_GENERATOR}"` belong together. They are now split and if someone removes it from `CYCLONE_BUILD_ARGS` or reorders the args, this will break.

Reply:
Right, corrected